clippy_lints/src/cfg_not_test.rs

```diff
@@ -0,0 +1,62 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_ast::NestedMetaItem;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Disallows usage of any conditional compilation that always excludes code from test builds.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Exclude code from tests builds. Is against simplicity in testing, guarding excessive mocking anti-pattern.
+    ///
+    /// Can show a codebase with a 100% coverage while having untested code.
+    ///
+    /// ### Example
+    /// ```rust
+    /// // an important that is a pain to test but not testing it can actually be dangerous
+    ///
+    /// #[cfg(not(test))]
+    /// important_check();
+    /// ```
+    ///
+    /// You should instead don't exclude any specific code from your test builds.
```

Comment:
```suggestion
    /// ### What it does
    /// Checks for usage of `cfg` that excludes code from `test` builds. (i.e., `#{cfg(not(test))]`)
    ///
    /// ### Why is this bad?
    /// This may give the false impression that a codebase has 100% coverage, yet actually has untested code.
    ///
    /// ### Example
    /// ```rust
    /// #[cfg(not(test))]
    /// important_check(); // I'm not actually tested, but not including me will falsely increase coverage!
    /// ```
    /// Use instead:
    /// ```rust
    /// important_check();
    /// ```
```